src/StoreApi/Formatters/FormatterInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\StoreApi\Formatters;
+
+/**
+ * FormatterInterface.
+ *
+ * @internal This API is used internally by Blocks--it is still in flux and may be subject to revisions.
+ */
+interface FormatterInterface {
+	/**
+	 * Format a given value and return the result.
+	 *
+	 * @param mixed $value Value to format.
+	 * @return mixed
+	 */
+	public function format( $value );
+
+	/**
+	 * Gets a formatter option.
+	 *
+	 * @param string $name Option name.
+	 * @param string $default Default returned if the option is not set.
+	 * @return string
+	 */
+	public function get_option( $name, $default = '' );
+
+	/**
+	 * Sets a formatter option.
+	 *
+	 * @param string $name Option name.
+	 * @param string $value Option value.
+	 */
+	public function set_option( $name, $value );
```

Comment:
I see this interface is intended to allow a way for changing how a formatter works. One potential problem I see with the current implementation is that any code implementing a formatter could not rely on what is set by default and would explicitly need to always set any configuration options so I wonder if this interface is needed. Should we instead just have `format` receive a second argument for any formatter configuration (for on the fly config) It could either be an instance of some sort of configuration interface or a simple array (and default to empty)?

Reply:
Well if we're not lazy instantialising we can do that instead. So long as the methods have the same signature across all formatters it will be ok.

Reply:
Mike's comment above should clear this because no one is sharing the same instance.

Reply:
It does 👍 (as noted in my response to the comment).